service/src/main/java/bio/terra/axonserver/app/controller/GcpResourceController.java

```diff
@@ -174,4 +178,47 @@ public ResponseEntity<ApiUrl> getDataprocClusterComponentUrl(
     String componentUrl = getCluster(workspaceId, resourceId).getComponentUrl(componentKey);
     return new ResponseEntity<>(new ApiUrl().url(componentUrl), HttpStatus.OK);
   }
+
+  /**
+   * Get a Dataproc cluster's metadata.
+   *
+   * @param workspaceId Terra Workspace ID
+   * @param resourceId Terra Resource ID
+   * @return cluster metadata
+   */
+  @Override
+  public ResponseEntity<ApiClusterMetadata> getDataprocClusterMetadata(
+      UUID workspaceId, UUID resourceId) {
+    ClusterConfig clusterConfig = getCluster(workspaceId, resourceId).getClusterConfig();
+
+    // Build metadata response body
+    ApiClusterMetadata metadata =
+        new ApiClusterMetadata()
+            .configBucket(clusterConfig.getConfigBucket())
+            .tempBucket(clusterConfig.getTempBucket())
+            .metadata(clusterConfig.getGceClusterConfig().getMetadata())
+            .managerNodeConfig(
+                DataprocMetadataBuilderUtils.buildInstanceGroupConfig(
+                    clusterConfig.getMasterConfig()))
+            .primaryWorkerConfig(
+                DataprocMetadataBuilderUtils.buildInstanceGroupConfig(
+                    clusterConfig.getWorkerConfig()))
+            .secondaryWorkerConfig(
+                DataprocMetadataBuilderUtils.buildInstanceGroupConfig(
+                    clusterConfig.getSecondaryWorkerConfig()))
+            .lifecycleConfig(
+                DataprocMetadataBuilderUtils.buildLifecycleConfig(
+                    clusterConfig.getLifecycleConfig()));
+
+    Optional.ofNullable(clusterConfig.getAutoscalingConfig())
```

Comment:
nit: set InitializationActions before AutoscalingConfig (same order of field definition in openapi)